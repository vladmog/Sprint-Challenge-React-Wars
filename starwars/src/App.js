import React, { useState, useEffect } from 'react';
import './App.css';
import axios from 'axios';
import People from './components/People';
import Header from './components/Header';

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.
  const [people, setPeople] = useState([]);
  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.

  useEffect(() => {
    axios.get('https://swapi.co/api/people/')
      .then (res => setPeople(res.data.results) //why can't I console.log before this setPeople?
      )
      .catch (err => console.log(err))
  }, [])

  console.log("STATE:", people)
  return (
    <div className="App">
      <Header />
      <People people = {people} />
    </div>
  );
}

export default App;
