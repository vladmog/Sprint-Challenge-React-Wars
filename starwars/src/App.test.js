import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});