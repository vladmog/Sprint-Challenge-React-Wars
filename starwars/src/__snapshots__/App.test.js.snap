// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <div>
    <h1
      className="Header"
    >
      React Wars
    </h1>
  </div>
  <div
    className="sc-bxivhb pvEHR"
  />
</div>
`;
